utilities/src/lib.rs

```diff
@@ -17,3 +17,60 @@
 pub mod signals;
 
 pub use signals::*;
+
+use std::path::{Path, PathBuf};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct NodeConfigDir {
+    path: PathBuf,
+
+    /// Use the legacy storage format?
+    /// This will eventuall be removed when the legacy storage format is no longer supported.
+    legacy: bool,
+
+    /// Is the node a development node?
+    /// This will eventually be removed when the development storage format is no longer supported.
+    dev: Option<u16>,
+}
+
+impl NodeConfigDir {
+    pub fn new(path: PathBuf) -> Self {
+        Self { path, legacy: false, dev: None }
+    }
+
+    pub fn new_test(dev: Option<u16>) -> Self {
+        if let Some(dev) = dev {
+            Self { path: PathBuf::from(format!(".node-config-test-{dev}")), legacy: false, dev: Some(dev) }
+        } else {
+            Self { path: PathBuf::from(".node-config-test"), legacy: false, dev: None }
+        }
+    }
+
+    pub fn new_legacy(path: PathBuf) -> Self {
+        Self { path, legacy: true, dev: None }
+    }
+
+    pub fn new_legacy_dev(path: PathBuf, dev: u16) -> Self {
+        Self { path, legacy: true, dev: Some(dev) }
+    }
+
+    pub fn new_development(dev: u16) -> Self {
+        Self { path: PathBuf::from(format!(".node-config-dev-{dev}")), legacy: false, dev: Some(dev) }
+    }
```

Comment:
That's a lot of options, would this be more succinct and easier to read if we pass in `StorageMode` directly?

Reply:
I also considered having a unified `StorageMode` instead and have it refer to the parent directory of the ledger directory instead, e.g. `~/.aleo/storage/` or `~/.aleo/storage-0` instead of `~/.aleo/storage/ledger-0`. That would be a breaking change, though, but make things easier in the long run.

To clarify, the breaking change would be that when you pass a custom storage path, snarkOS would then append `ledger` to it for the rocksDB storage, which does not happen in the current implementation.